src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/EditSavesModal.tsx

```diff
@@ -0,0 +1,153 @@
+
+  const initialValues = pick(collection, "name")
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+      .required(t("collectorSaves.editListModal.fields.name.required")) // TODO: confirm copy
+      .max(MAX_NAME_LENGTH),
+  })
+
+  const { submitMutation } = useUpdateCollection()
+
+  const { sendToast } = useToasts()
+
+  const handleSubmit = async (
+    formikValues: FormikValues,
+    formikHelpers: FormikHelpers<FormikValues>
+  ) => {
+    try {
+      await submitMutation({
+        variables: {
+          input: {
+            id: collection.internalID,
+            name: formikValues.name,
+          },
+        },
+        rejectIf: res => {
+          const result = res.updateCollection?.responseOrError
+
+          return result?.__typename === "UpdateCollectionFailure"
+            ? result.mutationError
+            : false
+        },
+      })
+
+      sendToast({
+        variant: "success",
+        message: t("collectorSaves.editListModal.success"),
+      })
+    } catch (error) {
+      logger.error(error)
+      sendToast({
+        variant: "error",
+        message: error.message ?? t("common.errors.somethingWentWrong"),
+      })
+    } finally {
+      formikHelpers.setSubmitting(false)
+      setIsEditModalOpen(false)
+    }
+  }
+
+  return (
+    <Formik<FormikValues>
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      onSubmit={handleSubmit}
+    >
+      {({
+        values,
+        errors,
+        touched,
+        handleBlur,
+        handleChange,
+        isSubmitting,
+      }) => {
+        return (
+          <ModalDialog
+            title={t("collectorSaves.editListModal.title")}
+            onClose={() => {
+              setIsEditModalOpen(false)
+            }}
+            width={["100%", 700]}
+          >
+            <Form /* TODO: confirm implicit form submission */>
+              <LabeledInput
+                name="name"
+                value={values.name}
+                title={t("collectorSaves.editListModal.fields.name.label")}
+                label={<EditIcon />}
+                error={touched.name && errors.name}
+                maxLength={MAX_NAME_LENGTH}
+                required // TODO: confirm
+                onChange={handleChange}
+                onBlur={handleBlur}
+              />
+              <Spacer y={1} />
+              <Text variant="xs">
+                {t(
+                  "collectorSaves.createNewListModal.remainingCharactersCount",
+                  { count: MAX_NAME_LENGTH - values.name.length }
+                )}
+              </Text>
+              <Spacer y={2} />
+              <Flex
+                justifyContent={["space-between"]}
+                flexDirection={["column", "row-reverse"]}
+              >
+                <Button type="submit" mb={[1, 0]} loading={!!isSubmitting}>
+                  {t("collectorSaves.editListModal.save")}
+                </Button>
```

Comment:
**suggestion**: Disable the "Save" button until the user has changed the name. We can use `dirty` value from formik for this (like [here](https://github.com/artsy/force/blob/a6331ebbe2687a36d87a848fc585db7167f6c878/src/Components/AuthDialog/Views/AuthDialogForgotPassword.tsx#L52))

Reply:
Added in 34534d6e97